src/h5web/visualizations/heatmap/IndexAxis.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Axis } from '@vx/axis';
+import { format } from 'd3-format';
+import styles from './HeatmapVis.module.css';
+import { IndexAxisProps, getOrientedAxisProps } from './utils';
+
+function IndexAxis(props: IndexAxisProps): JSX.Element {
+  const commonTickLabelProps = {
+    className: styles.tick,
+    textAnchor: 'inherit' as const,
+    dominantBaseline: 'inherit' as const,
+  };
+  const { axisProps, tickLabelProps } = getOrientedAxisProps(props);
+
+  const { orientation, numberPixels } = props;
+  return (
+    <svg className={styles.axis} data-orientation={orientation}>
+      <Axis
+        orientation={orientation}
+        strokeWidth={4}
+        tickFormat={format('0')}
+        numTicks={Math.min(numberPixels, 5)}
+        {...axisProps}
```

Comment:
I think having the two props here, each with its own ternary somewhere inside, would be a lot simpler. You could put the orientation condition into a const, like `isHorizAxis` to make the ternary shorter.